dan.chiniara/add troubleshooting for missing notifications from email (#23619)
DataDog/documentation: layouts/shortcodes/notifications-email.md

* Notify an active Datadog user by email with `@<DD_USER_EMAIL_ADDRESS>`. 

>     **Note**: An email address associated with a pending Datadog user invitation or a disabled user is considered inactive and does not receive notifications. Blocklists, ip or domain filtering, spam filtering, or email security tools may also be the reason notifications are missing. 

Reviewer: ```suggestion
    **Note**: An email address associated with a pending Datadog user invitation or a disabled user is considered inactive and does not receive notifications. Blocklists, IP or domain filtering, spam filtering, or email security tools may also cause missing notifications. 
```
